__tests__/ncnewsapp.test.js

```diff
@@ -87,20 +87,80 @@ describe('/api/articles', () => {
+          expect(article).toHaveProperty('title', expect.any(String));
+          expect(article).toHaveProperty('topic', expect.any(String));
+          expect(article).toHaveProperty('author', expect.any(String));
+          expect(article).toHaveProperty('created_at', expect.any(String));
+          expect(article).toHaveProperty('votes', expect.any(Number));
+          expect(article).toHaveProperty('article_img_url', expect.any(String));
+          expect(article).toHaveProperty('comment_count', expect.any(String));
+
+        })
       });
   })
+  test('GET:200 the articles should be returned in descending order of created_at ', () => {
+    return request(app)
+      .get('/api/articles')
+      .expect(200)
+      .then((response) => {
+        expect(response.body.articles).toBeSortedBy("created_at", { descending: true });
+      });
+    });
 });
+describe('/api/articles/:article_id/comments', () => {
+  test('200: Responds with an array of comments for the given article_id of which each comment should have the specified properties', () => {
+    return request(app)
+      .get('/api/articles/1/comments')
+      .expect(200)
+      .then((response) => {
+        const { comments } = response.body;
+        expect(comments).toHaveLength(11)
+        expect(Array.isArray(comments)).toBe(true);
+        comments.forEach((comment) => {
+          expect(comment).toHaveProperty('comment_id');
+          expect(comment).toHaveProperty('author');
+          expect(comment).toHaveProperty('article_id');
+          expect(comment).toHaveProperty('votes');
+          expect(comment).toHaveProperty('created_at');
+          expect(comment).toHaveProperty('body');
+        });
+      });
+  });
 
+test('200: Comments should be served with the most recent comments first. ', () => {
+  return request(app)
+    .get('/api/articles/3/comments')
+    .expect(200)
+    .then((response) => {
+      expect(response.body.comments).toBeSortedBy("created_at", { descending: true });
+    });
+  });
 
-
+  test('GET:404 sends an appropriate error message when given a valid but non-existent id', () => {
+    return request(app)
+      .get('/api/articles/999')
+      .expect(404)
+      .then((response) => {
+        expect(response.body.message).toBe('article does not exist');
+      });
+  });
+  test('GET:400 sends an appropriate error message when given an invalid id', () => {
+    return request(app)
+      .get('/api/articles/banana/comments')
+      .expect(400)
+      .then((response) => {
+        expect(response.body.message).toBe('Invalid id');
+      });
+  });
+  test('GET:200 sends an appropriate error message when given a valid article ID but article has no comments ', () => {
+    return request(app)
+      .get('/api/articles/2/comments')
+      .expect(200)
+      .then((response) => {
+        expect(response.body.message).toBe('article has no comments');
+      });
+  });
+});
```

Comment:
The response we would actually be expecting here would be an empty array - if you ever want to display a message like this to the user, you would handle this in the front end :-)